### PR TITLE
Feature/per question commenting

### PR DESCRIPTION
**Pull request by joefitter**

### CODE REVIEW

File: pages/project-version/middleware/index.js [hunk 1]
```diff
@@ -7,6 +9,110 @@ const getVersion = () => (req, res, next) => {
...
+          comments.map(({ comment, createdAt, changedBy: { firstName, lastName } }) => {
+            return {
+              comment,
+              // we want to show the date of the following status change, not the comment submission.
+              createdAt: ([...statuses].reverse().find(s => s.createdAt > createdAt) || {}).createdAt,
+              author: `${firstName} ${lastName}`,
+              isNew: isNew(req.user, data.activityLog, createdAt)
+            };
+          })
+            .reverse()
```
Comment (lennym) [inline pages/project-version/middleware/index.js]: Is there a reason this is not part of the chain? It's a bit odd there being one call which is one level lower.

File: pages/project-version/middleware/index.js [hunk 2]
```diff
@@ -7,6 +9,110 @@ const getVersion = () => (req, res, next) => {
...
+      const statuses = data.activityLog.filter(e => e.eventName.match(/^status:/));
+      // if most recent status change is returned to applicant (asru user) or
+      // resubmitted (non-asru), ignore comments made after this status change.
+      const ignoreAfter = statuses[0] && statuses[0].event.status === status && statuses[0].createdAt;
+      const comments = chain(data.activityLog)
+        .filter(e => e.eventName === 'comment')
+        .filter(e => !ignoreAfter || e.createdAt < ignoreAfter)
+        .groupBy(comment => comment.event.meta.payload.meta.field)
+        .mapValues(comments =>
+          comments.map(({ comment, createdAt, changedBy: { firstName, lastName } }) => {
```
Comment (lennym) [inline pages/project-version/middleware/index.js]: `{}` pls.

**Comment by lennym** [comment]
I'm 👌 with this, but it has my code in it, so I might let someone else click the button.

File: pages/project-version/middleware/index.js (hunk 1 above)
Comment (joefitter) [reply pages/project-version/middleware/index.js]: yes

Comment (joefitter) [reply pages/project-version/middleware/index.js]: the chain at this point is an object, the values are the arrays which need reversing

**Pull request merged**